packages/odata-service-writer/src/index.ts

```diff
@@ -133,11 +108,10 @@ async function generate(basePath: string, service: OdataService, fs?: Editor): P
     if (service.metadata) {
         // copy existing `ui5.yaml` as starting point for ui5-mock.yaml
         if (paths.ui5Yaml && ui5Config) {
-            const ui5MockConfig = await UI5Config.newInstance(ui5Config.toString());
-            ui5MockConfig.addMockServerMiddleware(service.path, await generateODataAnnotationConfig(basePath, fs));
-            fs.write(join(dirname(paths.ui5Yaml), 'ui5-mock.yaml'), ui5MockConfig.toString());
-
-            // also add mockserver middleware to ui5-local.yaml
+            const webappPath = await getWebappPath(basePath);
```

Comment:
The flow of execution here is causing the changes to the package.json's, if you move this code after the ui5LocalConfig.addMockServerMiddleware, you should see less file differences.